### DISCUSSION TITLE
WATER_IN / OUT not supported in LHM for ASUS motherboards...

### DISCUSSION THREAD

**Discussion by EyoSama** [LibreHardwareMonitor/LibreHardwareMonitor#611] [Q&A]
In LHM I do not see the WATER_IN and WATER_OUT sensor readings for my ASUS ROG Maximus XIII Hero motherboard. Is this something that can be added to LHM?

![image](https://user-images.githubusercontent.com/92912797/144658777-6a6e8309-2a4b-472f-8f79-aa74ce831d1c.png)

I use TTController from MoshiMoshi, which uses LHM. TTController allows controlling the Thermaltake Fans and LEDs, and does an incredible job at it (a much better job than the actual Thermaltake software, which is unusable). The combination of TTController and LHM, is simply amazing and very impressive. Kudos to everyone involved!

But, one thing that would make it even more functional, would be the ability to be able to control the Fans based on the Water Temperature of a Liquid Cooling loop, since it makes much more sense to have the Fans react to the water temperature and not the individual GPU and CPU Temps.

Cheers!

**Comment by EyoSama** [comment]
Heya, just wondering if anyone is available to answe?

Thanks.

**Comment by EyoSama** [comment]
It's been a year, was just wondering if anyone had an answer to this? How can we see the Water_In / Out temperatures on ASUS motherboards?

Thanks.

**Reply by WinFinn** [reply]
Looks like your MB is not specifically added in LHM. And values you see are from the default setup for ASUS Nuvoton controller. 
Looking at other series 500 motherboard it looks the values you are looking for is on the ASUS EC controller. You could use HWinfo to verify this, where those values will be listed under the ASUS EC section. 
LHM also supports the EC sensor, but in general it require a lot of reverse engineering and testing with your specific MB to implement.

**Reply by EyoSama** [reply]
Thanks for the response, Win, much appreciated. Do you have any guidance on where I could get information on doing the last thing you said? I do have HWInfo.

**Reply by WinFinn** [reply]
Not really. Best I found was following a link provided in a comment under the LHM Embedded Controller section. (There is corresponding linux driver). The linux driver had some guidance.
However W_in and W_out looks same between the latest board families, so you might have luck in just testing that out.. If you are known to coding you can fork LHM and test.
